test/fixture.rb

```diff
@@ -56,7 +62,7 @@ module Wrapper
     class BeforeAfterA
       def before_a(number)
         Fixtures.callstack.push([self.class, __method__, number])
-        number + 5
+        { args: [number + 5], kwargs: {} }
```

Comment:
This is the only breaking change I introduced in the PR. IMO we need to separate kwargs from positional args when returning values from `before_ methods` in wrappers.

Reply:
hmm so the before wrappers now have to always return this hash structure? even when they don't use kwargs or even more than a single value as return?

While this makes it "work" it kinda breaks the whole idea that these wrappers are transparent and don't have to know such things and the return of this is directly passed to the next wrapper without some internal gem transformation. This was a core design idea behind how these wrappers work and are most of the reason behind the rather complicated meta programming inside the gem. IMHO if this is removed I'd probably rethink how this whole parameter passing things works and not try to just make it somehow work without a nice interface.